src/serializer/Referentializer.js

```diff
@@ -207,12 +229,14 @@ export class Referentializer {
         invariant(residualBinding !== undefined);
         if (residualBinding.modified) {
           // Initialize captured scope at function call instead of globally
+          if (!residualBinding.declarativeEnvironmentRecord) residualBinding.referentialized = true;
           if (!residualBinding.referentialized) {
             if (!shouldReferentializeInstanceFn(instance)) {
               // TODO #989: Fix additional functions and referentialization
               throw new FatalError("TODO: implement referentialization for prepacked functions");
```

Comment:
Throwing a FatalError without logging a CompilerDiagnostics will always result in a bad user experience.